[VoiceRSS] Fix SAT findings and other code improvements

wborn opened the pull request:
Several improvements for the VoiceRSS TTS addon based on the recent review comments of the Google (#3721) and Polly (#3222) TTS voice addon PRs.

cweitkamp commented on addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:
```diff
@@ -147,37 +146,37 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
-        HashSet<Voice> voices = new HashSet<Voice>();
+    private Set<Voice> initVoices() {
+        Set<Voice> voices = new HashSet<>();
         Set<Locale> locales = voiceRssImpl.getAvailableLocales();
```
While you are editing this anyway we can change this too. There is no need to assign it to a variable but rather use it in the foreach loop directly:

for(Locale locale : voiceRssImpl.getAvailableLocales()) {
...
}

cweitkamp commented on addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:
```diff
@@ -147,37 +146,37 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
-        for (Locale local : locales) {
-            Set<String> voiceLabels = voiceRssImpl.getAvailableVoices(local);
+        for (Locale locale : locales) {
+            Set<String> voiceLabels = voiceRssImpl.getAvailableVoices(locale);
```
see above

cweitkamp commented on addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:
```diff
@@ -147,37 +146,37 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
-        HashSet<AudioFormat> audioFormats = new HashSet<AudioFormat>();
+    private Set<AudioFormat> initAudioFormats() {
+        Set<AudioFormat> audioFormats = new HashSet<>();
         Set<String> formats = voiceRssImpl.getAvailableAudioFormats();
```
see above

cweitkamp commented on addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:
```diff
@@ -76,34 +76,34 @@ protected void activate(Map<String, Object> config) {
     @Override
     public Set<Voice> getAvailableVoices() {
-        return this.voices;
+        return voices;
```
Should we maybe return an unmodifiable set here?

cweitkamp commented on addons/voice/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:
```diff
@@ -76,34 +76,34 @@ protected void activate(Map<String, Object> config) {
     @Override
     public Set<AudioFormat> getSupportedFormats() {
-        return this.audioFormats;
+        return audioFormats;
```
Should we maybe return an unmodifiable set here?

wborn commented:
I've addressed all your review comments @cweitkamp!

Now that the number of voice addons is growing it makes sense for ESH to provide classes or may be even a service to simplify working with TTS voice caches. WDYT @kaikreuzer?
